src/sparsezoo/v2/objects/model_directory.py

```diff
@@ -377,32 +395,56 @@ def _file_from_files(
 
         if not files_found:
             return None
-        if len(files_found) == 1:
+        elif len(files_found) == 1:
+            return files_found[0]
+
+        elif display_name == "model.onnx" and len(files_found) == 2:
+            # `model.onnx` file may be found twice:
+            #   - directly in the root directory
+            #   - inside `deployment` directory
+            # if this is the case, return the first file
+            # (the other one is a duplicate)
```

Comment:
maybe add `assert files_found[0] == files_found[1]` to be sure?

Reply:
+1 for raising `ValueError` with check proposed by @corey-nm

Reply:
Well, `ValueError` here is actually pretty useless. `files_found[0] == files_found[1]` should be just the condition of the `if` statement.

Reply:
After taking a second look, I do not think that this assertion is really needed.